File: android/app/src/main/kotlin/com/example/flutter_application_3/SessionReplayHolder.kt
package com.example.flutter_application_3

import android.app.Application
import android.content.Context
import com.amplitude.android.sessionreplay.SessionReplay

object SessionReplayHolder {
    @Volatile private var sr: SessionReplay? = null

    fun init(
        context: Context,
        apiKey: String,
        deviceId: String,
        sessionId: Long,
        sampleRate: Double = 1.0,
        eu: Boolean = false,
        enableRemoteConfig: Boolean = false
    ) {
        if (sr != null) return
        sr = SessionReplay(
            apiKey = apiKey,
            context = context.applicationContext,
            deviceId = deviceId,
            sessionId = sessionId,       // en milisegundos epoch
            sampleRate = sampleRate,     // durante QA puedes usar 1.0 (100%)
            enableRemoteConfig = enableRemoteConfig
        )
    }

    fun setSessionId(sessionId: Long) { sr?.setSessionId(sessionId) }
    fun setDeviceId(deviceId: String) { sr?.setDeviceId(deviceId) }

    /** Devuelve el mapa con "[Amplitude] Session Replay ID" listo para adjuntar al evento */
    fun getProperties(): Map<String, Any> = sr?.getSessionReplayProperties() ?: emptyMap()

    fun flush() { sr?.flush() }
}

File: ios/Runner/SessionReplayHolder.swift
// ios/Runner/SessionReplayHolder.swift
import Foundation
import AmplitudeSessionReplay

final class SessionReplayHolder {
  static let shared = SessionReplayHolder()
  private init() {}

  private var sr: SessionReplay?
  private var lastSessionId: Int64?

  func initIfNeeded(
    apiKey: String,
    deviceId: String,
    sessionId: Int64,
    sampleRate: Float = 1.0, // <- Debe ser Float
    eu: Bool = false,
    enableRemoteConfig: Bool = true
  ) {
    guard sr == nil else { return }

    // Uso correcto del enum
    let zone: ServerZone = eu ? .EU : .US

    sr = SessionReplay(
      apiKey: apiKey,
      deviceId: deviceId,
      sessionId: sessionId,      // epoch ms
      sampleRate: sampleRate,    // Float
      serverZone: zone,          // Enum, no String
      enableRemoteConfig: enableRemoteConfig
    )
    sr?.start()
    lastSessionId = sessionId
  }

  func setSessionId(_ sessionId: Int64) {
    guard let sr else { return }
    if lastSessionId != sessionId {
      sr.sessionId = sessionId
      lastSessionId = sessionId
    }
  }

  func setDeviceId(_ deviceId: String) {
    sr?.deviceId = deviceId
  }

  func getProperties() -> [String: Any] {
    sr?.additionalEventProperties ?? [:]
  }

  func flush() {
    sr?.flush()
  }
}